hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java

```diff
@@ -274,12 +274,13 @@ public void run() {
         public void run() {
             IdleStrategy idlerLocal = idlerCooperative;
             long idleCount = 0;
+            boolean shouldLogFinest = logger.isFinestEnabled();
```

Comment:
Logging level can be changed dynamically. As a compromise, we could call `isFinestEnabled` once per iteration of tasklets. Also the name can be `logFinestEnabled`.

Reply:
What about this? I had issues in the past when a production system got into a bad state and I wanted to enable logging at runtime, but didn't work because the code extracted the value in similar way...